nixpkgs_review/review.py

```diff
@@ -472,9 +489,24 @@ def filter_packages(
     system: str,
     allow: AllowedFeatures,
     nix_path: str,
+    build_passthru_tests: bool,
 ) -> set[str]:
     packages: set[str] = set()
 
+    # reduce number of times the passthru.tests are evaluated, by either doing
+    # it here or in join_packages
+    if build_passthru_tests and len(specified_packages) == 0:
+        changed_attrs = package_attrs(
+            changed_packages,
+            system,
+            allow,
+            nix_path,
+            build_passthru_tests=build_passthru_tests,
+        )
+        changed_packages = set(
+            changed_attrs[path].name for path in changed_attrs.keys()
```

Comment:
Sorry for the late reply. I think because of the potential large memory consumption of nixos tests, we need to evaluate them isolation as they will otherwise make the build potentially go out of memory, i.e. if you try to have a large package set rebuild.
Therefore I would suggest to evaluate those packages ahead of time i.e. using `nix-instantiate` and add them to the final `nix-build`. Of course a better design would be some sort of pipe-lining where already evaluated packages gets build as the rest of the evaluation is still in progress. But this probably requires bigger refactorings and could implemented at a later point similar to nix-fast-build.

Reply:
Maybe at first we could do this way but warn about this possible problem and then address it in a follow up PR